docs/blog-post.md

```diff
@@ -0,0 +1,286 @@
+file. It can be used for detecting accidental changes of events (journal) and states (snapshots) with a simple `diff`.
+
+To dump persistence schema for each sbt module where `AkkaSerializationHelperPlugin` is enabled, run:
+
+```shell
+sbt ashDumpPersistenceSchema
+```
+
+It saves created dump into a file (default is `target/<sbt-module-name>-dump-persistence-schema-<version>.yaml`)
+
+#### Example dump
+```yaml
+- name: org.random.project.Data
+  typeSymbol: trait
+- name: org.random.project.Data.ClassTest
+  typeSymbol: class
+  fields:
+  - name: a
+    typeName: java.lang.String
+  - name: b
+    typeName: scala.Int
+  - name: c
+    typeName: scala.Double
+  parents:
+  - org.random.project.Data
+- name: org.random.project.Data.ClassWithAdditionData
+  typeSymbol: class
+  fields:
+  - name: ad
+    typeName: org.random.project.Data.AdditionalData
+  parents:
+  - org.random.project.Data
+```
+
+Then, simple `diff` command can be used to check the difference between the version of a schema from `develop`/`main` branch and the
+version from the current commit.
+
+<img width="1043" alt="easy-to-diff" src="https://user-images.githubusercontent.com/49597713/172350313-cd03f033-ca68-40c3-a4df-064456c483d0.png">
+
+### 3. Jackson Akka Serializer drawbacks
+
+Using Jackson Serializer for akka-persistence is also one of the pitfalls. Akka Serialization Helper provides an alternative - a more Scala-friendly serializer that uses [Circe](https://circe.github.io/circe/).
+
+Examples below show some problems that might occur when combining Jackson with Scala code:
+
+#### Example 1 
+Dangerous code for Jackson:
+
+```scala
+case class Message(animal: Animal) extends MySer
+
+sealed trait Animal
+
+final case class Lion(name: String) extends Animal
+final case class Tiger(name: String) extends Animal
+```
+This seems to be all right, but unfortunately will not work with Jackson serialization - in runtime there will be an exception with message like: "Cannot construct instance of `Animal`(...)" - as abstract types need to be mapped to concrete types explicitly in code. So, to make this code work, a lot of Jackson annotations should be added:
+
+```scala
+case class Message(animal: Animal) extends MultiDocPrintService
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+@JsonSubTypes(
+  Array(
+    new JsonSubTypes.Type(value = classOf[Lion], name = "lion"),
+    new JsonSubTypes.Type(value = classOf[Tiger], name = "tiger")))
+sealed trait Animal
+
+final case class Lion(name: String) extends Animal
+final case class Tiger(name: String) extends Animal
+
+```
+
+#### Example 2
+
+If a Scala object is defined:
+```scala
+case object Tick
+```
+
+There **will not be exceptions** during serialization but Jackson will create
+another instance of `Tick` instead of restoring the `object Tick`'s underlying singleton. So, deserialization will end up very bad...
```

Comment:
> another instance of `Tick`

Actually... of `Tick` or `Tick$`? https://virtuslab.github.io/tips/#scala-and-java/scala-objects-under-the-hood

Reply:
Changed description to:
```
There **will not be exceptions** during serialization but, during deserialization, Jackson will create
another `Tick` object instead of restoring the `object Tick`'s underlying singleton. So, deserialization will end up very bad...
```
You are right - scala compiler creates Tick$ class so we can say that the object is an instance of type `Tick$`. However, I think that sing "Tick$" class name for this example will make it too mind blowing (complicated) for people who are not Scala experts. Let me know, if we can leave it like that.